src/parser/retail/modules/items/dragonflight/UsurpedFromBeyond.tsx

```diff
@@ -0,0 +1,273 @@
+      // Default heroic itemlevel as fallback to not implode
+      431;
+
+    const buffAmount = voiceOfTheSilentStarStealAmount(itemLevel);
+
+    return -buffAmount;
+  }
+
+  private currentLowestSecondaryStat(): SECONDARY_STAT {
+    return [
+      {
+        stat: SECONDARY_STAT.CRITICAL_STRIKE,
+        value: this.deps.statTracker.currentCritRating,
+      },
+      {
+        stat: SECONDARY_STAT.HASTE,
+        value: this.deps.statTracker.currentHasteRating,
+      },
+      {
+        stat: SECONDARY_STAT.MASTERY,
+        value: this.deps.statTracker.currentMasteryRating,
+      },
+      {
+        stat: SECONDARY_STAT.VERSATILITY,
+        value: this.deps.statTracker.currentVersatilityRating,
+      },
+    ].reduce((acc, stat) => {
+      if (stat.value < acc.value) {
+        return stat;
+      }
+      return acc;
+    }).stat;
+  }
+
+  private updateStats(
+    stat: SECONDARY_STAT,
+    amount: number,
+    event: ApplyBuffEvent | RemoveBuffEvent,
+  ) {
+    this.deps.statTracker.forceChangeStats(
+      {
+        [stat]: amount,
+      },
+      event,
+    );
+  }
+
+  private thiefTable() {
+    const thiefMap = this.thefts.reduce(
+      (acc: Record<number, { duration: number; count: number }>, { thief, end, start }) => {
+        if (thief) {
+          if (!acc[thief]) {
+            acc[thief] = {
+              duration: 0,
+              count: 0,
+            };
+          }
+
+          acc[thief].duration +=
+            (end ?? this.owner.fight.end_time) - (start ?? this.owner.fight.start_time);
+          acc[thief].count += 1;
+        }
+        return acc;
+      },
+      {},
+    );
+
+    const victimEntries = Object.entries(thiefMap)
+      .sort(([, a], [, b]) => b.duration - a.duration)
+      .map(([victimID, { duration, count }]) => ({
+        victimID: Number(victimID),
+        duration,
+        count,
+      }));
+
+    const combatants = this.deps.combatants.getEntities();
+
+    return (
+      <table className="table table-condensed">
+        <thead>
+          <tr>
+            <th style={{ textAlign: 'left' }}>Thief</th>
+            <th>Duration</th>
+            <th>Count</th>
+          </tr>
+        </thead>
+        <tbody>
+          {victimEntries.map(({ victimID, duration, count }) => {
+            const combatant = combatants[victimID];
+            const className = classColor(combatant);
+
+            return (
+              <tr key={victimID}>
+                <td style={{ textAlign: 'left' }}>
+                  <span className={className}>{combatant.name}</span>
+                </td>
+                <td>{formatDuration(duration)}</td>
+                <td>{count}</td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+    );
+  }
+
+  statistic() {
+    if (this.thefts.length === 0) {
+      // This module is always active, but if never robbed, should not show any statistic
```

Comment:
we do get all players' combatantinfo and cloaks cannot be swapped in combat so you could do `this.active = this.combatantInfoEvents.some(hasCloak);`

Reply:
Aah that's a great idea! I've added a check that _another_ player has cloak and only activate in that case!